monix-reactive/jvm/src/main/scala/monix/reactive/compress/internal/operators/InflateOperator.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2014-2020 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive.compress.internal.operators
+
+import java.util.zip.{DataFormatException, Inflater}
+import java.{util => ju}
+
+import monix.execution.Ack
+import monix.execution.Ack.{Continue, Stop}
+import monix.reactive.Observable.Operator
+import monix.reactive.compress.CompressionException
+import monix.reactive.observers.Subscriber
+
+import scala.annotation.tailrec
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+private[compress] final class InflateOperator(bufferSize: Int, noWrap: Boolean)
+  extends Operator[Array[Byte], Array[Byte]] {
+
+  def apply(out: Subscriber[Array[Byte]]): Subscriber[Array[Byte]] =
+    new Subscriber[Array[Byte]] {
+      implicit val scheduler = out.scheduler
+
+      private[this] var isDone = false
+      private[this] var ack: Future[Ack] = _
+      private[this] val inflater = new InflateAdapter(bufferSize, noWrap)
+
+      def onNext(elem: Array[Byte]): Future[Ack] = {
+        if (isDone) {
+          Stop
+        } else {
+          try {
+            val result = inflater.onChunk(elem)
+            // signaling downstream
+            ack = out.onNext(result)
+            ack
+          } catch {
+            case e if NonFatal(e) =>
+              onError(e)
+              Stop
+          }
+        }
+      }
+
+      def onError(ex: Throwable): Unit =
+        if (!isDone) {
+          isDone = true
+          inflater.close()
+          out.onError(ex)
+        }
+
+      def onComplete(): Unit =
+        if (!isDone) {
+          isDone = true
+          if (ack == null) ack = Continue
+          ack.syncOnContinue {
+            try {
+              out.onNext(inflater.finish())
+            } catch {
```

Comment:
Similar issue as `GunzipOperator` - it could call both `onError` and `onComplete`

Reply:
I explained in this comment: https://github.com/monix/monix/pull/1244/files#r495495768